core/src/main/java/io/substrait/relation/Expand.java

```diff
@@ -51,8 +54,8 @@ public static ImmutableExpand.ConsistentField.Builder builder() {
   public abstract static class SwitchingField implements ExpandField {
     public abstract List<Expression> getDuplicates();
 
-    public Type getType() {
-      return getDuplicates().get(0).getType();
+    public Stream<Type> getTypes() {
+      return getDuplicates().stream().map(Expression::getType);
```

Comment:
maybe I'm misunderstanding something here - but are you sure this is right? The [Substrait spec comments](https://github.com/substrait-io/substrait/blob/2e13d0b7b4a2b5ac8221e7f02cc8f04456c8f11c/proto/substrait/algebra.proto#L500) say 
> All duplicates must return the same type class but may differ in nullability.

so it doesn't seem right to take a list of the duplicates' types.